res/res_respoke_session.c

```diff
@@ -620,6 +621,8 @@ struct respoke_session *respoke_session_create(
 		return NULL;
 	}
 
+	session->metadata = ast_json_ref(metadata);
```

Comment:
Add a NULL check for metadata (it looks like it can be null) as ast_json_ref is not null tolerant.